GeoJSON.js

```diff
@@ -1,239 +1,246 @@
+            } else {
+                obj = [];
+                for (i = 0; i < geojson.features.length; i++) {
+                    obj.push(_geometryToGoogleMaps(geojson.features[i].geometry, opts, geojson.features[i].properties));
+                }
+            }
+            break;
+
+        case "GeometryCollection":
+            if (!geojson.geometries) {
+                obj = _error("Invalid GeoJSON object: GeometryCollection object missing \"geometries\" member.");
+            } else {
+                obj = [];
+                for (i = 0; i < geojson.geometries.length; i++) {
+                    obj.push(_geometryToGoogleMaps(geojson.geometries[i], opts));
+                }
+            }
+            break;
+
+        case "Feature":
+            if (!(geojson.properties && geojson.geometry)) {
+                obj = _error("Invalid GeoJSON object: Feature object missing \"properties\" or \"geometry\" member.");
+            } else {
+                obj = _geometryToGoogleMaps(geojson.geometry, opts, geojson.properties);
+            }
+            break;
+
+        case "Point": case "MultiPoint": case "LineString": case "MultiLineString": case "Polygon": case "MultiPolygon":
+            if (geojson.coordinates) {
+                obj = _geometryToGoogleMaps(geojson, opts);
+            } else {
+                obj = _error("Invalid GeoJSON object: Geometry object missing \"coordinates\" member.");
+            }
+
+            break;
+
+        default:
+            obj = _error("Invalid GeoJSON object: GeoJSON object must be one of \"Point\", \"LineString\", \"Polygon\", \"MultiPolygon\", \"Feature\", \"FeatureCollection\" or \"GeometryCollection\".");
     }
-    return newObj;
-  };
-		
-	var obj;
-	
-	var opts = options || {};
-	
-	switch ( geojson.type ){
-	
-		case "FeatureCollection":
-			if (!geojson.features){
-				obj = _error("Invalid GeoJSON object: FeatureCollection object missing \"features\" member.");
-			}else{
-				obj = [];
-				for (var i = 0; i < geojson.features.length; i++){
-					obj.push(_geometryToGoogleMaps(geojson.features[i].geometry, opts, geojson.features[i].properties));
-				}
-			}
-			break;
-		
-		case "GeometryCollection":
-			if (!geojson.geometries){
-				obj = _error("Invalid GeoJSON object: GeometryCollection object missing \"geometries\" member.");
-			}else{
-				obj = [];
-				for (var i = 0; i < geojson.geometries.length; i++){
-					obj.push(_geometryToGoogleMaps(geojson.geometries[i], opts));
-				}
-			}
-			break;
-		
-		case "Feature":
-			if (!( geojson.properties && geojson.geometry )){
-				obj = _error("Invalid GeoJSON object: Feature object missing \"properties\" or \"geometry\" member.");
-			}else{
-				obj = _geometryToGoogleMaps(geojson.geometry, opts, geojson.properties);
-			}
-			break;
-		
-		case "Point": case "MultiPoint": case "LineString": case "MultiLineString": case "Polygon": case "MultiPolygon":
-			obj = geojson.coordinates
```

Comment:
I changed the ternary operator to an if-then-else - it was impacting readability.

Also there was a bug here:

    obj = geojson.coordinates ? obj = _geometry...

(There was an extra assignment within the ternary operator, causing it to break)